libgnucash/engine/qofclass.cpp

```diff
@@ -327,20 +237,16 @@ find_reference_param_cb(QofParam *param, gpointer user_data)
     {
         return;
     }
-    b->list = g_list_append(b->list, param);
+    auto list = static_cast<GList*>(user_data);
+    list = g_list_prepend (list, param);
 }
 
 GList*
 qof_class_get_referenceList(QofIdTypeConst type)
 {
-    GList *ref_list;
-    struct param_ref_list b;
-
-    ref_list = NULL;
-    b.list = NULL;
-    qof_class_param_foreach(type, find_reference_param_cb, &b);
-    ref_list = g_list_copy(b.list);
-    return ref_list;
+    GList *list = nullptr;
+    qof_class_param_foreach(type, find_reference_param_cb, &list);
+    return g_list_reverse (list);
```

Comment:
This is silly, use `std::foreach` directly and replace `find_reference_param_cb` with a lambda. Three's no point in using C wrappers inside a C++ file.